### DISCUSSION TITLE
Error catching

### DISCUSSION THREAD

**Discussion by ArnyminerZ** [leikoilja/ha-google-home#116] [Ideas]
Now that the integration is going one step further, maybe we should use some kind of error catcher, so we can know the user's problems before they even open a new issue, and get full detailed logs.

I've been using [Sentry](https://sentry.io/) in Android for a while, and it works like a charm, and there's also a [Python API](https://docs.sentry.io/platforms/python/).

Maybe this isn't appropiate, since we are a lot of people, and recollecting data may be an issue, but since all the data is anonymous, maybe we can inform the user, and give access to the errors logs to just the project's maintainers or something, what do you think?

**Comment by leikoilja** [comment]
Uuuu that one is interesting. Are you using `developer` free plan? 🤔 
I'm unsure if we can put something like that in the HomeAssistant 😅

**Reply by ArnyminerZ** [reply]
Yeah, I'm using the free plan, since my user base is not that large, but now that I think about it, it might be not enough for us...

**Reply by DurgNomis-drol** [reply]
@ArnyminerZ I love your idea! 😄  

But seeing how angry the HA community got when they enabled PWND checks in HAOS. I don't think we should do this, even if it is anonymous. 😢

**Reply by leikoilja** [reply]
I mean if we would ever get to the consensus to implement it - it could always be just a checkbox "I allow .... to make the integration better" on the config flow

**Reply by ArnyminerZ** [reply]
Yeah, I see that it could be the way to go @leikoilja maybe we set it by default to false and everybody is happy 😄

**Reply by DurgNomis-drol** [reply]
@ArnyminerZ @leikoilja If we can do that I think it is a great idea.

Maybe a stupid question, but if it is disabled by default, do you think that people will use it then? or should it be a dialog box style question so the user gets prompt? 😄

**Reply by ArnyminerZ** [reply]
Maybe when the integration is installed, just after logging in, we can explain and give the user the possibility to enable the functionality.

**Reply by DurgNomis-drol** [reply]
Great idea! And if we are lucky 5K errors on the free plan will be enough if it is optional 😄

**Reply by ArnyminerZ** [reply]
Yeah, and if not, simply Sentry will block the new calls until a new month start, so no unwanted charges 🎉

**Comment by leikoilja** [comment]
@KapJI have you thoughts on that?

**Reply by KapJI** [reply]
I agree this may be useful but many people don't like their data being collected.

As you mentioned we add do an opt-out setting in config flow ("Send anonymous error reports to help us to make this integration better") with being explicit about what is actually collected.

**Comment by leikoilja** [comment]
I haven't worked much with logging and error-catching systems like that before, but are there any free alternatives? Maybe there are some open source solutions that might fit us?

**Reply by KapJI** [reply]
I've seen Sentry in few other projects, it's quite popular choice.

**Comment by leikoilja** [comment]
@ArnyminerZ do you have interest at creating an issue for that and implementing the feature? :) 
If so, we can first create an issue(Edit: #124) regarding `config_flow` that it should allow optional checkbox "Send anonymous error reports to help us to make this integration better" and allow users to enter their `master_token` straight. And the following issue (blocked by 1st) regarding `Sentry` implementation. 
If you would be up for a `Sentry`, i can take the `config_flow` issue

**Reply by leikoilja** [reply]
@ArnyminerZ? 🤔
